KS-5-4 задача 4

VVShchegolev opened the pull request.

Commit:
KS-5-4 задача 4

ievetrovdev commented on src/main/kotlin/Lesson_5_Ex_4/Lesson_5_Ex_4.kt:
```diff
@@ -0,0 +1,32 @@
+
+    if (username != REGISTERED_USERNAME) {
+        println("Пользователь не зарегистрирован. Пожалуйста, зарегистрируйтесь.")
+        return
```
Сейчас проверка на имя пользователя и пароль сделана поочередно с выходом из функции, если пользователь не найден. Это рабочий подход, но предлагаю попробовать другое решение: делать проверку пароля внутри блока, где имя пользователя уже подтверждено. Это делает логику более последовательной: система сначала проверяет имя, и только потом — пароль. При этом не нужно выходить из функции раньше времени. Такой стиль ближе к тому, как часто пишется код в реальных проектах — и полезно будет его освоить.

VVShchegolev committed:
Update KS_5_4 Задача 4 исправлена

Что Исправлено: 
Теперь проверка пароля внутри блока, где имя пользователя уже подтверждено.

ievetrovdev commented on src/main/kotlin/Lesson_5_Ex_4/Lesson_5_Ex_4.kt:
```diff
@@ -0,0 +1,34 @@
+            "но вы, наверное, об этом не заботитесь. [вздыхает ещё глубже...] " +
+            "Да, вперед, пожалуйста, вводите свои данные...")
+    println("Введите имя пользователя:")
+    val username = readLine()!!
```
Оператор `!!` (утверждение “это не `null`”) использовать не рекомендуется. При получении null, программа упадет с ошибкой. Подробнее об этом будет в дальнейших уроках [https://youtu.be/hf4vrHucYNU](https://youtu.be/hf4vrHucYNU) 
Сама IDEA подсказывает использовать современный безопасный readln().

ievetrovdev commented on src/main/kotlin/Lesson_5_Ex_4/Lesson_5_Ex_4.kt:
```diff
@@ -0,0 +1,34 @@
+
+    val message = if (username == REGISTERED_USERNAME) {
+        println("Введите пароль:")
+        val password = readLine()!!
```
И здесь.

VVShchegolev committed:
Update KS-5-4 Задание 4

Изменил readLine() на readln()

ievetrovdev reviewed (approved):
Отлично!
